airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -262,8 +266,8 @@ def stream_slices(
             start_date = stream_state.get(self.cursor_field)
             if start_date:
                 start_date = pendulum.from_format(start_date, ReportStream.REPORT_DATE_FORMAT, tz="UTC")
-                # We already processed records for date specified in stream state, move to the day after
-                start_date += timedelta(days=1)
+                # Amazon ads updates the data for the next 3 days
+                start_date += timedelta(days=-3)
```

Comment:
start_date represents date when sync had been run. You better edit date shifting calculation in the _calc_report_generation_date method.

Reply:
@avida so this is like look back we need to pull data for last 3 days

Reply:
@harshithmullapudi I got this. We already have calculation of date shift in _calc_report_generation_date